homeassistant/components/bosch_shc/switch.py

```diff
@@ -0,0 +1,234 @@
+
+    for switch in session.device_helper.smart_plugs_compact:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["smartplugcompact"],
+            )
+        )
+
+    for switch in session.device_helper.camera_eyes:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["cameraeyes"],
+            )
+        )
+
+    for switch in session.device_helper.camera_360:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["camera360"],
+            )
+        )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class SHCSwitch(SHCEntity, SwitchEntity):
+    """Representation of a SHC switch."""
+
+    entity_description: SHCSwitchEntityDescription
+
+    def __init__(
+        self,
+        device: SHCDevice,
+        parent_id: str,
+        entry_id: str,
+        description: SHCSwitchEntityDescription,
+    ) -> None:
+        """Initialize a SHC switch."""
+        super().__init__(device, parent_id, entry_id)
+        self.entity_description = description
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return (
+            getattr(self._device, self.entity_description.on_key)
+            == self.entity_description.on_value
+        )
+
+    def turn_on(self, **kwargs) -> None:
+        """Turn the switch on."""
+        setattr(self._device, self.entity_description.on_key, True)
+
+    def turn_off(self, **kwargs) -> None:
+        """Turn the switch off."""
+        setattr(self._device, self.entity_description.on_key, False)
+
+    def toggle(self, **kwargs) -> None:
+        """Toggle the switch."""
+        setattr(self._device, self.entity_description.on_key, not self.is_on)
+
+    @property
+    def should_poll(self) -> bool:
+        """Switch needs polling."""
+        return self.entity_description.should_poll
+
+    def update(self) -> None:
+        """Trigger an update of the device."""
+        self._device.update()
+
+
+class SHCRoutingSwitch(SHCEntity, SwitchEntity):
+    """Representation of a SHC routing switch."""
+
+    def __init__(self, device: SHCDevice, parent_id: str, entry_id: str) -> None:
+        """Initialize an SHC communication quality reporting sensor."""
+        super().__init__(device, parent_id, entry_id)
+        self._attr_name = f"{device.name} Routing"
+        self._attr_unique_id = f"{device.serial}_routing"
+        self._attr_entity_category = EntityCategory.CONFIG
+        self._attr_icon = "mdi:wifi"
```

Comment:
Icon and Entity Category here should be moved to class attributes (outside of __init__ method)